File: 2493-reverse-odd-levels-of-binary-tree/2493-reverse-odd-levels-of-binary-tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
  func dfs(left *TreeNode, right *TreeNode, l int) {
        if left == nil || right == nil {
            return
        }
        if l & 1 != 0 {
            t := left.Val
            left.Val = right.Val
            right.Val = t
        }
        dfs (left.Left, right.Right, l + 1)
        dfs (left.Right, right.Left, l + 1)
        return
    }
func reverseOddLevels(root *TreeNode) *TreeNode {
   

    dfs(root.Left, root.Right, 1)
    return root 
}

File: 2493-reverse-odd-levels-of-binary-tree/2493-reverse-odd-levels-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
typedef struct TreeNode Node;

void swap(int* x, int* y) {
    *x = *x + *y;
    *y = *x - *y;
    *x = *x - *y;
}
void change(Node* left, Node* right, int l) {
    if (!left || !right)
        return;
    if (l & 1) {
        swap(&left->val, &right->val);
    }
    change(left->left, right->right, l + 1);
    change(left->right, right->left, l + 1);
    return;
}
struct TreeNode* reverseOddLevels(struct TreeNode* root) {
    change(root->left, root->right, 1);
    return root;
}